src/plugins/expressions/common/execution/execution_contract.ts

```diff
@@ -38,18 +40,21 @@ export class ExecutionContract<Input = unknown, Output = unknown, InspectorAdapt
    * This function never throws.
    */
   getData = async (): Promise<Output | ExpressionValueError> => {
-    try {
-      return await this.execution.result;
-    } catch (e) {
-      return {
-        type: 'error',
-        error: {
-          name: e.name,
-          message: e.message,
-          stack: e.stack,
-        },
-      };
-    }
+    return this.execution.result
+      .pipe(
+        take(1),
```

Comment:
why are we taking only the first result ?

Reply:
That's not the first result but the latest. This is because `.toPromise()` is resolved only when the subject is complete. Since we never complete `input$`, we have to limit it here so the returned promise will resolve with the latest result.